pkg/controller/csr/csr_controller.go

```diff
@@ -90,15 +111,21 @@ func (r *ReconcileCSR) Reconcile(ctx context.Context, request reconcile.Request)
 		return reconcile.Result{}, nil
 	}
 
-	clusterName := getClusterName(csr)
-	cluster := clusterv1.ManagedCluster{}
-	err = r.clientHolder.RuntimeClient.Get(ctx, types.NamespacedName{Name: clusterName}, &cluster)
-	if errors.IsNotFound(err) {
-		// no managed cluster, do nothing.
-		return reconcile.Result{}, nil
+	// Check if any approval condition matches
+	shouldApprove := false
+	for _, condition := range r.approvalConditions {
```

Comment:
@zhujian7 
The `checks` is renamed to `appprovalConditions` to avoid misunderstanding.